### PR TITLE
Support array result include sequence action

### PR DESCRIPTION
**Pull request by ningyougang**
Depend on below prs: 
- https://github.com/apache/openwhisk/pull/5290
- https://github.com/apache/openwhisk-runtime-go/pull/170

- [x] Support array result
  - [x] make rust runtime to support array result for common action
  - [x] make rust runtime to support array result for sequence action (support array as input param)

### CODE REVIEW

**Comment by ningyougang** [comment]
The rust supports sequence action by default, test steps as below
* Write split.rs and sort.rs
```rust
extern crate serde_json;

use serde_derive::{Deserialize, Serialize};
use serde_json::{Error, Value};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
struct Input {
    #[serde(default = "stranger")]
    name: String,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
struct Output {
    greeting: String,
}

fn stranger() -> String {
    "stranger".to_string()
}

pub fn main(args: Value) -> Result<Value, Error> {
    // has no split, just return array, need to receive this value in next action(sort)
    let output = ["a", "b"];
    serde_json::to_value(output)
}

extern crate serde_json;

use serde_derive::{Deserialize, Serialize};
use serde_json::{Error, Value};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
struct Input {
    #[serde(default = "stranger")]
    name: String,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
struct Output {
    greeting: String,
}

fn stranger() -> String {
    "stranger".to_string()
}

pub fn main(args: Value) -> Result<Value, Error> {
    println!("param args is {}", args);

    let inputParam = args.as_array();
    println!("input param as array is {:?}", inputParam);

    let defaultOutput = ["finally-a", "finally-b"];
    match inputParam {
       None => serde_json::to_value(defaultOutput),
       Some(x) => serde_json::to_value(x),
    }
}
```
* Create the action and invoke it
```shell
wsk -i action create /whisk.system/utils/split-rust --kind rust:1.34 ~/split.rs
wsk -i action create /whisk.system/utils/sort-rust --kind rust:1.34 ~/sort.rs
wsk -i action create mySequence-rust --sequence /whisk.system/utils/split-rust,/whisk.system/utils/sort-rust
wsk -i action invoke --result mySequence-rust -r -v
```

refer to: https://docs.serde.rs/serde_json/value/enum.Value.html

File: core/rust1.34/Dockerfile [hunk 1]
```diff
@@ -31,12 +31,12 @@ RUN curl -sL \
...
   | tar xzf -\
   && cd openwhisk-runtime-go-*/main\
-  && GO111MODULE=on go build -o /bin/proxy
+  && GO111MODULE=on CGO_ENABLED=0 go build -o /bin/proxy
 
 FROM rust:1.34
 
 # select the builder to use
-ARG GO_PROXY_BUILD_FROM=release
+ARG GO_PROXY_BUILD_FROM=source
```
Comment (ningyougang) [inline core/rust1.34/Dockerfile]: Need to change it to `release` due to https://github.com/apache/openwhisk-runtime-java/pull/140#discussion_r939689436

Comment (ningyougang) [reply core/rust1.34/Dockerfile]: Already changed to `release`

**Review by dgrove-oss — APPROVED** [review]
LGTM

**Pull request merged**